File: dcp-solutions/2024-10-28/solution.go
```markdown
package main

import (
    "fmt"
)

func maxSubArray(nums []int) int {
    // Handle empty array
    if len(nums) == 0 {
        return 0
    }

    // Initialize maximum sum and current sum to negative infinity
    max_sum := int(-1e9)
    current_sum := 0

    // Iterate over the array, updating current sum at each step
    for _, num := range nums {
        current_sum = max(num, current_sum+num)

        // Update max sum if current sum is greater
        if current_sum > max_sum {
            max_sum = current_sum
        }
    }

    return max_sum
}

func main() {
}
```

File: dcp-solutions/2024-10-28/solution.php
```php
function maxSubArray($nums) {
    if (empty($nums)) {
        return 0;
    }

    $max_sum = $current_sum = $nums[0];

    foreach ($nums as $num) {
        $current_sum = max($num, $current_sum + $num);
        $max_sum = max($max_sum, $current_sum);
    }

    return $max_sum;
}
```